WordPress/src/main/java/org/wordpress/android/ui/stories/StoryMediaSaveUploadBridge.kt

```diff
@@ -0,0 +1,170 @@
+ * and then only create the post and upload it.
+ */
+class StoryMediaSaveUploadBridge @Inject constructor(
+    private val addLocalMediaToPostUseCase: AddLocalMediaToPostUseCase,
+    private val savePostToDbUseCase: SavePostToDbUseCase,
+    private val uploadService: UploadServiceFacade,
+    private val networkUtils: NetworkUtilsWrapper,
+    private val postUtils: PostUtilsWrapper,
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher
+) : CoroutineScope, LifecycleObserver, EditorMediaListener {
+    // region Fields
+    private var job: Job = Job()
+
+    override val coroutineContext: CoroutineContext
+        get() = mainDispatcher + job
+
+    @Inject lateinit var editPostRepository: EditPostRepository
+
+    @Suppress("unused")
+    @OnLifecycleEvent(ON_CREATE)
+    fun onCreate(source: LifecycleOwner) {
+        EventBus.getDefault().register(this)
+    }
+
+    @Suppress("unused")
+    @OnLifecycleEvent(ON_DESTROY)
+    fun onDestroy(source: LifecycleOwner) {
+        // note: not sure whether this is ever going to get called if we attach it to the lifecycle of the Application
+        // class, but leaving it here prepared for the case when this class is attached to some other LifeCycleOwner
+        // other than the Application.
+        cancelAddMediaToEditorActions()
+        EventBus.getDefault().unregister(this)
+    }
+
+    // region Adding new composed / processed frames to a Story post
+    private fun addNewStoryFrameMediaItemsToPostAndUploadAsync(site: SiteModel, saveResult: StorySaveResult) {
+        // let's invoke the UploadService and enqueue all the files that were saved by the FrameSaveService
+        val frames = StoryRepository.getStoryAtIndex(saveResult.storyIndex).frames
+        val uriList = frames.map { Uri.fromFile(it.composedFrameFile) }
+        addNewMediaItemsToPostAsync(site, uriList)
+    }
+
+    private fun addNewMediaItemsToPostAsync(site: SiteModel, uriList: List<Uri>) {
+        // this is similar to addNewMediaItemsToEditorAsync in EditorMedia
+        launch {
+            addLocalMediaToPostUseCase.addNewMediaToEditorAsync(
+                    uriList,
+                    site,
+                    freshlyTaken = false, // we don't care about this
+                    editorMediaListener = this@StoryMediaSaveUploadBridge,
+                    doUploadAfterAdding = true
+            )
+            postUtils.preparePostForPublish(editPostRepository.getEditablePost()!!, site)
+            savePostToDbUseCase.savePostToDb(WordPress.getContext(), editPostRepository, site)
+
+            if (networkUtils.isNetworkAvailable()) {
+                postUtils.trackSavePostAnalytics(
+                        editPostRepository.getPost(),
+                        site
+                )
+                uploadService.uploadPost(WordPress.getContext(), editPostRepository.id, true)
```

Comment:
Can we inject this context somehow so that we avoid using the static method call?

Reply:
Been checking this, I think we need to make a context factory so it's injectable. I've seen sometimes we use `init()` to pass some variables to ViewModels and other injectable classes in WPAndroid, so went that way for now in 85bda8b237